### PR TITLE
pushing ticket 8

### PR DESCRIPTION
**Pull request by yos244**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -186,46 +215,74 @@ describe("Error handling", () => {
...
-    .send(comment)
-    .expect(400)
-    .then((response)=>{
-      expect(response.body.msg).toEqual("Invalid id")
-    })
+      .post("/api/reviews/2/comments")
+      .send(comment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toEqual("Invalid id");
```
Comment (pkaodev) [inline __tests__/app.test.js:245]: maybe an error message other than "invalid id" would make more sense here

File: __tests__/app.test.js [hunk 2]
```diff
@@ -186,46 +215,74 @@ describe("Error handling", () => {
...
+  });
+  test("PATCH: 400 empty object", () => {
+    const obj = {};
+    return request(app)
+      .patch("/api/reviews/5")
+      .send(obj)
+      .expect(400)
+      .then((response)=>{
+        expect(response.body.msg).toEqual(`Invalid data type`)
+      })
```
Comment (pkaodev) [inline __tests__/app.test.js:286]: make separate tests for missing username/body to make sure each is being handled

File: model/model.js [hunk 3]
```diff
@@ -144,3 +143,36 @@ RETURNING *;
         });
     });
 };
+
+exports.editVotes = ({ review_id }, voteInc) => {
+  if (isNaN(voteInc)) {
+    return Promise.reject({ status: 400, msg: `Invalid data type` });
+  }
```
Comment (pkaodev) [inline model/model.js:150]: this isn't needed.  If you try to make a query with an invalid data type Postgres will throw an error automatically which you can handle.

File: model/model.js [hunk 4]
```diff
@@ -144,3 +143,36 @@ RETURNING *;
...
+  return db
+    .query(
+      `
+  SELECT * FROM reviews
+  `
+    )
+    .then((allReviews) => {
+      if (review_id > allReviews.rows.length) {
+        return Promise.reject({ status: 400, msg: `Invalid id` });
+      }
```
Comment (pkaodev) [inline model/model.js:160]: We can refactor this a bit to be more clear.

Do Select all from reviews where review_id = (the id being sent from the client)

you can then reject the promise if it returns no results (the review doesnt exist).

**Comment by pkaodev** [comment]
It seems you have done changes for multiple tickets here which has made it a bit confusing, some of my comments are concerning older tickets.

looks good, just a couple of things:
-see comments 
-add test for inc_votes of -1
-test for missing username key

Message me if you want to discuss any of my comments.  Please submit another PR when you have made these changes.

File: model/model.js (hunk 3 above)
Comment (yos244) [reply model/model.js:150]: but it does not

**Pull request merged**